src/commands/dev.ts

```diff
@@ -200,6 +200,57 @@ export async function command(commandOptions: CommandOptions) {
   const messageBus = new EventEmitter();
   const mountedDirectories: [string, string][] = [];
 
+  onProcessExit(() => {
+    hmrEngine.disconnectAllClients();
+  });
+
+  console.log = (...args) => {
+    messageBus.emit('CONSOLE', {level: 'log', args});
+  };
+  console.warn = (...args) => {
+    messageBus.emit('CONSOLE', {level: 'warn', args});
+  };
+  console.error = (...args) => {
+    messageBus.emit('CONSOLE', {level: 'error', args});
+  };
+
+  // Start painting immediately, so we can surface errors & warnings to the
+  // user, and they can watch the server starting up. Search for ”SERVER_START”
+  // for the actual start event below.
+  paint(messageBus, config.scripts, undefined, {
+    addPackage: async (pkgName: string) => {
+      isLiveReloadPaused = true;
+      messageBus.emit('INSTALLING');
+      currentlyRunningCommand = execa(
+        isYarn(cwd) ? 'yarn' : 'npm',
+        isYarn(cwd) ? ['add', pkgName] : ['install', '--save', pkgName],
+        {
+          env: npmRunPath.env(),
+          extendEnv: true,
+          shell: true,
+          cwd,
+        },
+      );
+      currentlyRunningCommand.stdout.on('data', (data) => process.stdout.write(data));
+      currentlyRunningCommand.stderr.on('data', (data) => process.stderr.write(data));
+      await currentlyRunningCommand;
+      currentlyRunningCommand = installCommand(commandOptions);
+      await currentlyRunningCommand;
+      await updateLockfileHash(DEV_DEPENDENCIES_DIR);
+      await cacache.rm.all(BUILD_CACHE);
+      inMemoryBuildCache.clear();
+      currentlyRunningCommand = null;
+
+      dependencyImportMap = JSON.parse(
+        await fs
+          .readFile(dependencyImportMapLoc, {encoding: 'utf-8'})
+          .catch(() => `{"imports": {}}`),
+      );
+      messageBus.emit('INSTALL_COMPLETE');
+      isLiveReloadPaused = false;
+    },
+  });
```

Comment:
Notice that now `addPackage` is the only command. The other options have been moved to the `SERVER_START` message bus event